### DISCUSSION TITLE
Basic types

### DISCUSSION THREAD

**Discussion by data-man** [c3lang/c3c#164] [Q&A] [answered]
According to https://github.com/c3lang/c3docs/blob/master/docs/types.md#basic-types
The simple `types.c3`:
```c

//import std::cinterop; (types.c3:1) Error: No module named 'std::cinterop' could be found, did you type the name right?

func int main()
{
    bool b;
    char c;
//    byte b2; (types.c3:8) Error: Expected ';'.
    short s;
    ushort u;
    int i;
    uint ui;
    long l;
    ulong ul;
    iptr ip;
    uptr up;
    iptrdiff ipd;
    uptrdiff upd;
    isize is;
    usize us;

//    half h; Error: Expected ';'.
    float f;
    double d;
//    quad q; Error: Expected ';'.
//    f16 f_16; Error: Expected ';'.
//    f32 f_32; Error: Expected ';'.
//    f64 f_64; Error: Expected ';'.

    return 0;
}
```

**Comment by lerno** [comment, marked as answer]
quad and half support isn't complete yet. I am not sure whether I'll support the type aliases (e.g. f16 etc) as aliases.
`byte` / `char` now is `char` and `ichar` respectively. I'll fix the docs for that. I'll create the cinterop package when I have the $switch working 100%, because the source code will look something like:

```c
$switch (CINTSIZE):
   $case 8:
     define CInt = ichar;
   $case 16:
     define CInt = short;
   $case 32:
     define CInt = int;
   $case 64:
     define CInt = long;
$endswitch     
```

.. in that file. So that's why it's delayed.

**Reply by goyalyashpal** [reply]
these seem to be done now... right?  so, this discussion can be closed?
